test/e2e/aws_test.go

```diff
@@ -135,11 +147,393 @@ var _ = Describe("functional tests", func() {
+
+			By("Creating the MachineDeployment")
+			createMachineDeployment(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, mdBootstrapConfig, instanceType, initialReplicas)
+
+			By("Creating the LB service")
+			elbName := createLBService(metav1.NamespaceDefault, lbServiceName, clusterK8sClient)
+			verifyElbExists(elbName, true)
+
+			By("Deleting the Cluster")
+			deleteCluster(namespace, clusterName)
+
+			By("Verifying whether provisioned LB deleted")
+			verifyElbExists(elbName, false)
+		})
+	})
+
+	Describe("Provisioning volumes dynamically and retain on cluster deletion", func() {
+		nginxStatefulsetInfo := statefulSetInfo{
+			name:                      "nginx-statefulset",
+			namespace:                 metav1.NamespaceDefault,
+			replicas:                  int32(2),
+			selector:                  map[string]string{"app": "nginx"},
+			storageClassName:          "aws-ebs-volumes",
+			volumeName:                "nginx-volumes",
+			svcName:                   "nginx-svc",
+			svcPort:                   int32(80),
+			svcPortName:               "nginx-web",
+			containerName:             "nginx",
+			containerImage:            "k8s.gcr.io/nginx-slim:0.8",
+			containerPort:             int32(80),
+			podTerminationGracePeriod: int64(30),
+			volMountPath:              "/usr/share/nginx/html",
+		}
+
+		It("It should create volumes and volumes should not be deleted along with cluster infra", func() {
+			By("Creating a cluster with single control plane")
+			clusterK8sClient := makeSingleControlPlaneCluster(namespace, clusterName, awsClusterName, cpAWSMachinePrefix, cpBootstrapConfigPrefix, cpMachinePrefix, instanceType, testTmpDir)
+
+			By("Creating the MachineDeployment")
+			createMachineDeployment(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, mdBootstrapConfig, instanceType, initialReplicas)
+
+			By("Deploying StatefulSet on infra")
+			createStatefulSet(nginxStatefulsetInfo, clusterK8sClient)
+			awsVolIds := getVolumeIds(nginxStatefulsetInfo, clusterK8sClient)
+			verifyVolumesExists(awsVolIds)
+
+			By("Deleting the Cluster")
+			deleteCluster(namespace, clusterName)
+
+			By("Verifying dynamically provisioned volumes retention")
+			verifyVolumesExists(awsVolIds)
+
+			By("Deleting retained dynamically provisioned volumes")
+			deleteRetainedVolumes(awsVolIds)
+		})
+	})
+
+	Describe("MachineDeployment with invalid subnet ID", func(){
+		It("It should be creatable and deletable", func(){
+			By("Creating a cluster with single control plane")
+			makeSingleControlPlaneCluster(namespace, clusterName, awsClusterName, cpAWSMachinePrefix, cpBootstrapConfigPrefix, cpMachinePrefix, instanceType, testTmpDir)
+
+			By("Creating Machine Deployment with invalid subnet ID")
+			subnetId := "notcreated"
+			createPendingMachineDeployment(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, mdBootstrapConfig, instanceType,nil, &subnetId, initialReplicas)
```

Comment:
Use `createMachineDeployment` func

Reply:
Here i am creating machine deployments with invalid configuration where the MachineDeployment will not come to running stat. Hence, can't use `createMachineDeployment` which will check MachineDeployment running or not and fail if not running.

Reply:
Ok. My bad. 👍